### PR TITLE
Add libuv/1.38.0

### PR DESCRIPTION
**Pull request by yipdw**
Specify library name and version:  **libuv/1.38.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.  **Recipe changes tested with Conan 1.25.2.**
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.  **Shared and static builds tested with clang 10 on Ubuntu 20.04; test package completed successfully and all hook checks pass.**

### CODE REVIEW

File: recipes/libuv/all/patches/1.38.0/fix-cmake.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (yipdw) [inline recipes/libuv/all/patches/1.38.0/fix-cmake.patch:1]: libuv's CMake code underwent a large cleaning in v1.35.0.  This patch appears to be a correct conversion of what 1.34.2's `fix-cmake.patch` did, but I would appreciate extra scrutiny on it.

File: recipes/libuv/all/patches/1.38.0/fix-cmake.patch [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+-    USING_UV_SHARED=1
+-  PRIVATE
+-    BUILDING_UV_SHARED=1
+-    ${uv_defines})
++add_library(uv ${uv_sources})
++get_target_property(target_type uv TYPE)
++if (target_type STREQUAL "SHARED_LIBRARY")
++  target_compile_definitions(uv
++    INTERFACE
++      USING_UV_SHARED=1
```
Comment (uilianries) [inline recipes/libuv/all/patches/1.38.0/fix-cmake.patch:21]: do you need to export USING_UV_SHARED as compiler definition?

Comment (yipdw) [reply recipes/libuv/all/patches/1.38.0/fix-cmake.patch:21]: Hmm.  The test package seems to build without it, but it seems like a good idea to fully emulate the `uv` target's interface in Conan.  I'll add that to the Conanfile.

Comment (yipdw) [reply recipes/libuv/all/patches/1.38.0/fix-cmake.patch:21]: `USING_UV_SHARED=1` added to `cpp_info.defines` in 965ebbd8d5eb7021ba15907868b440644c9a85b9.

**Pull request merged**